src/persist-txn/src/txns.rs

```diff
@@ -691,17 +680,6 @@ where
         let op = &self.metrics.compact_to;
         op.run(async {
             tracing::debug!("compact_to {:?}", since_ts);
-            // TODO(jkosh44) While this comment accurately reflects the
-            // current behavior of `self.txns_cache`, the current behavior
-            // is not correct. See https://github.com/MaterializeInc/materialize/issues/26893.
-            //
-            // This call to compact the cache only affects the write and
-            // registration times, not the unapplied batches. The unapplied batches
-            // have a very important correctness invariant to hold, are
-            // self-compacting as batches are applied, and are handled below. This
-            // means it's always safe to compact the cache past where the txns shard
-            // is physically compacted to, so do that regardless of min_unapplied_ts
-            // and of whether the maybe_downgrade goes through.
```

Comment:
This comment isn't really accurate anymore but I didn't really know what to replace it with. I think it also highlights that there are still a few open questions or improvements to make. Such as
- Now that both the unapplied batches and the write times contain all unapplied writes, why do we need both? I think we need both because we are able to hold the write times back from compacting while we cannot hold the unapplied batches back, but I'm not 100% convinced.
- Both `TxnsHandle` and `TxnsCacheState` have their own logic for holding back compaction for unapplied writes. Does that make sense? Should that be combined somehow?